### ISSUE TITLE
Regression: ``cabal test`` when no test suite is available fails

### ISSUE THREAD

**Issue by svenpanne** [haskell/cabal#3215]
Doing ``cabal test`` in a package without a test suite fails, see https://travis-ci.org/haskell-opengl/OpenGLRaw/jobs/114109606#L1022. This happens with hvr's cabal 1.24 package and is a regression, cabal 1.22 is happy in that case, see https://travis-ci.org/haskell-opengl/OpenGLRaw/jobs/114109605#L929.

**Comment by 23Skidoo** [comment]
This was introduced in 1ec1cc2dcbfbe2c46b1c27119284a180963e0f1a, when I changed a call to `notice` to a call to `die`, and the message first appeared in e2ec932330d82f5ff374f66cca0332087a8b87b0. This is easy to fix, but I'm not sure that this a bug to be honest. Why are you expecting a success exit code in this case?

**Comment by svenpanne** [comment]
First of all: This is a very ad hoc regression, causing lots of my Travis CI jobs to fail for no good reason (at least I see none). Of course one could hack around this, but the question is: Why? Why should failing in this case be better than succeeding? Incompatible changes should better have a very good reason, otherwise things get annoying quickly.

My reading of ``cabal test`` is: It succeeds when all exisiting tests succeed, which is trivially true if there are no tests. This is the same reasoning why ``and []`` returns ``True`` in Haskell. This is a) consistent and b) removes the need to handle the empty case specially. The same reasoning holds for the test suite(s).

In a nutshell: If it's easy, please change it back. And I'm not sure if it correct to simply remove the ``bug`` label from this issue, this *is* a bug IMHO.

**Comment by 23Skidoo** [comment]
> My reading of ``cabal test`` is: It succeeds when all exisiting tests succeed, which is trivially true if there are no tests. This is the same reasoning why ``and []`` returns ``True`` in Haskell.

OK, makes sense.

**Comment by svenpanne** [comment]
I'm happy that we agree now. :smile: Don't get me wrong: Personally I can hack my several .travis.yaml files relatively quickly, my real concern is widespread breakage in the Haskell ecosystem at large.
